### PR TITLE
Add security override for sha1

### LINKED ISSUE #2181: Add CLI parameter `--allow-sha1-key-sigs`
## Description

From today SHA-1 key signatures are marked as weak, we should add some CLI option to still allow it in case of user need.

### PR DESCRIPTION
**Pull request by desvxx**
fixes #2181

### CODE REVIEW

**Comment by ni4** [comment]
@desvxx Could you please rebase this on main, as PR with CI fixes is already merged? Thanks!

File: src/tests/cli_tests.py [hunk 1]
```diff
@@ -4066,6 +4067,33 @@ def test_allow_weak_hash(self):
...
+    def test_allow_sha1_key_sigs(self):
+        src, sig = reg_workfiles('cleartext', '.txt', '.sig')
+        random_text(src, 120)
+
+        ret, out, _ = run_proc(RNPK, ['--keyfile', data_path(PUBRING_7), '--notty', '--list-keys'])
+        self.assertEqual(ret, 0)
+        self.assertRegex(out, r'(?s)^.*\[INVALID\].*$')
+        ret, out, _ = run_proc(RNPK, ['--keyfile', data_path(PUBRING_7), '--notty', '--list-keys', '--allow-sha1-key-sigs'])
+        self.assertEqual(ret, 0)
+        self.assertRegex(out, r'(?s)^.*pub.*2024-06-03.*sub.*2024-06-03.*$')
```
Comment (ni4) [inline src/tests/cli_tests.py:4079]: Could you please add NotRegex with INVALID check here?

File: src/rnpkeys/rnpkeys.cpp [hunk 2]
```diff
@@ -142,6 +143,7 @@ struct option options[] = {
   {"set-expire", required_argument, NULL, OPT_SET_EXPIRE},
   {"current-time", required_argument, NULL, OPT_CURTIME},
   {"allow-weak-hash", no_argument, NULL, OPT_ALLOW_WEAK_HASH},
+  {"allow-sha1-key-sigs", no_argument, NULL, OPT_ALLOW_SHA1},
```
Comment (ni4) [inline src/rnpkeys/rnpkeys.cpp:146]: Could you please add a new option to the man files as well?

**Review by ni4 — CHANGES_REQUESTED** [review]
Overall LGTM, just few minor updates.

**Review by ni4 — APPROVED** [review]
LGTM, thanks!

**Review by maxirmx — APPROVED** [review]
LGTM

**Pull request merged**